src/cbc_sdk/endpoint_standard/base.py

```diff
@@ -368,3 +384,225 @@ def _search(self, start=0, rows=0):
                 log.debug("resetting total_results for this query to {0}".format(current))
                 self._total_results = current
                 break
+
+class EnrichedEventQuery(Query):
+    """Represents the query logic for an asychronous Event query.
+
+    This class specializes `Query` to handle the particulars of
+    enriched events querying.
+    """
+    def __init__(self, doc_class, cb):
+        super(EnrichedEventQuery, self).__init__(doc_class, cb)
+        self._sort_by = None
+        self._group_by = None
+        self._batch_size = 100
+        self._default_args = {}
+        self._query_token = None
+        self._timeout = 0
+        self._timed_out = False
+        self._sort = []
+        self._time_range = {}
+
+    def or_(self, **kwargs):
+        """ or_ criteria are explicitly provided to EnrichedEvent queries although they are endpoint_standard.
+            This method overrides the base class in order to provide or_() functionality rather than
+            raising an exception
+        """ 
+        self._query_builder.or_(None, **kwargs)
+        return self
+
+    def _get_query_parameters(self):
+        """Need to override base class implementation as it sets custom (invalid) fields"""
+        args = self._default_args.copy()
+        args['query'] = self._query_builder._collapse()
+
+        return args
+
+    def set_rows(self, rows):
+        """
+        Sets the 'rows' query body parameter, determining how many rows of results to request.
+        Args:
+            rows (int): How many rows to request.
+        """
+        if not isinstance(rows, int):
+            raise ApiError(f"Rows must be an integer. {rows} is a {type(rows)}.")
+        self._batch_size = rows
```

Comment:
I think we must distinguish between the total number of rows (parameter in the POST request to create the search job) to get, and the batch size (passed as a GET parameter in the /results requests to get results) for each retrieved page.
Here both are tied and if I use `set_rows(10000)` I may have an issue because I can request 10k total results, but will it be nice from the API to download 10k at once? There is no pagination anymore
It depends on how you prefer us to hit the API :)

Reply:
My suggestion would be to have a default batch size, of 50 or 100 for example, for page retrieval and perhaps keep it internal like in #69
Then expose a public `rows` parameter, as already done, which only allows to set the total number of results desired

Reply:
OK, switching to using the rows argument to the method for this. Also set the API defaults and we now check if we pass >10K API limit to set_rows.

Reply:
Good idea to check the max yes!

Reply:
Hum I'm still not very fan of the default of 10 but as you wish :)
I have a case where it sometimes uses 0 so it goes in an infinite loop... i'm trying to investigate

Reply:
Ok I found the issue, it comes from here:
https://github.com/carbonblack/carbon-black-cloud-sdk-python/blob/8e276bcc032115c66008e9faa3c7073806fb54e5/src/cbc_sdk/base.py#L724-L725
`numrows` is default so 0 which is passed to `EnrichedEventQuery._search`

Reply:
I am in a bit of confusion as to how to handle that. I'm thinking about introducing a check if it is zero and in that case, set it to API default. But this also doesn't look quite clean to me. Maybe introducing a second argument to set_rows to set the pagination batch size would be better. Still not sure. What do you think would be better and more intuitive?

Reply:
OK, I split that into set_rows() to set the total result rows and set_batch_size to set the batch size as per a paginated results call

Reply:
Don't add `set_batch_size` as this already exists on the paginated query object which your class inherits https://github.com/carbonblack/carbon-black-cloud-sdk-python/blob/8e276bcc032115c66008e9faa3c7073806fb54e5/src/cbc_sdk/base.py#L728

 As for the infinite loop, there shouldn't be an infinite loop with the following lines in the code as this will break the loop. https://github.com/carbonblack/carbon-black-cloud-sdk-python/blob/CBPAPI-2025_enriched_events/src/cbc_sdk/endpoint_standard/base.py#L632-L633

Reply:
You can also remove the following line https://github.com/carbonblack/carbon-black-cloud-sdk-python/blob/CBPAPI-2025_enriched_events/src/cbc_sdk/endpoint_standard/base.py#L398 as this is set by the super call to the parent inherited object here https://github.com/carbonblack/carbon-black-cloud-sdk-python/blob/CBPAPI-2025_enriched_events/src/cbc_sdk/endpoint_standard/base.py#L287

Reply:
Oops, didn't notice it has implementation in the base class...
Anyway, in the API documentation for that route they say the default value is 10 while in the Query class that we inherit it is initialized to 100. Still unsure what to do though as 10 is the default as per the REST API, but it doesn't necessarily need to be the CBC SDK default. Which would be better?

Reply:
I would go with 100 since the 10 for the API is the default for the UI

Reply:
OK, removing set_batch_size and defaulting to 10 in the constructor.

Reply:
No default to 100 we can be more optimal from a developer's perspective 10 is only needed for the UI

Reply:
I meant removing both the set_batch_size and defaulting to 10. Anyway, looks like I am too sleepy already, will review the changes and push tomorrow.

Reply:
OK, removed.

Reply:
FYI I just tested with the latest code as of now, and it works very nicely :)
I can use set_rows() to define how many results in total I want, and by default it fetches via pages of 100 rows which is nice!
Thank you to both of you!

One small remark perhaps: I noticed `self._rows = 500` but this default value is never used. If we don't call `set_rows()` then `self._default_args["rows"] = self._rows` is not called and thus no "rows" parameter is sent when launching the search job and so the server-side API's default is used :)

Reply:
Fixed.

Reply:
Confirmed :)
Default is now 500 rows in total, and no regression when using "set_rows()" 👍